src/rgw/support/rgw_support_bundle

```diff
@@ -166,8 +162,40 @@ class RGWSupportBundle:
         if coredumps:
             crash_dump_dir = os.path.join(log_dir, 'rgw_debug')
             if os.path.exists(crash_dump_dir):
-                shutil.copytree(crash_dump_dir,\
-                    os.path.join(RGWSupportBundle._tmp_src, 'rgw_debug'))
+                if size_limit:
+                    latest_5_cores = RGWSupportBundle._get_latest_files(
+                        directory=crash_dump_dir, number_of_files=5)
+                    for file_name in latest_5_cores:
+                        shutil.copyfile(os.path.join(crash_dump_dir, file_name),
+                            os.path.join(RGWSupportBundle._tmp_src, 'rgw_debug'))
+                else:
+                    shutil.copytree(crash_dump_dir,\
+                        os.path.join(RGWSupportBundle._tmp_src, 'rgw_debug'))
+
+        # remaining size limit in folder
+        remaining_size_limit = RGWSupportBundle._get_remaining_folder_size(
+            RGWSupportBundle._tmp_src, size_limit)
+        # apply truncate logic on all textual logs
+        if size_limit and not coredumps:
```

Comment:
so size limit is applied for textual logs only when core files not being collected, why?
If core dumps are asked, size_limit are not avoided, as logic says at line 165 above

Reply:
as stated [in discussion](https://jts.seagate.com/browse/CORTX-33037?focusedCommentId=2401289&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-2401289), 
```
4. should size filters be considered when collecting core dumps as they tend to be bulky in size?
As stated above, no. But the number of core dumps can be decided based on the available size
```
we choose to avoid truncating the files if core dumps are to be collected as they may get bulky and consume all the size limit, if that happens we will end up having all empty log files.
will work on limiting the cores to a max of 2 if the size limit permits else at least 1

Reply:
> we choose to avoid truncating the files if core dumps are to be collected as they may get bulky and consume all the size limit, if that happens we will end up having all empty log files.

Log files should have higher priority than core dumps, and not vice versa. 

> will work on limiting the cores to a max of 2 if the size limit permits else at least 1

this looks addressed in code above now

Reply:
made the amends to first consider log files, then cores.